sdk/identity/identity/src/credentials/azurePipelinesCredential.ts

```diff
@@ -118,27 +116,46 @@ export class AzurePipelinesCredential implements TokenCredential {
       logger.error(
         `${credentialName}: Authenticated Failed. Received null token from OIDC request. Response status- ${
           response.status
-        }. Complete response - ${JSON.stringify(response)}`,
+        }. Complete response - ${JSON.stringify(response)}`
       );
-      throw new CredentialUnavailableError(
+      throw new AuthenticationError(
+        response.status,
         `${credentialName}: Authenticated Failed. Received null token from OIDC request. Response status- ${
           response.status
-        }. Complete response - ${JSON.stringify(response)}`,
+        }. Complete response - ${JSON.stringify(response)}`
       );
     }
-    const result = JSON.parse(text);
-    if (result?.oidcToken) {
-      return result.oidcToken;
-    } else {
-      logger.error(
-        `${credentialName}: Authentication Failed. oidcToken field not detected in the response. Response = ${JSON.stringify(
-          result,
-        )}`,
-      );
-      throw new CredentialUnavailableError(
-        `${credentialName}: Authentication Failed. oidcToken field not detected in the response. Response = ${JSON.stringify(
-          result,
-        )}`,
+    try {
+      const result = JSON.parse(text);
+      if (result?.oidcToken) {
+        return result.oidcToken;
+      } else {
+        if (response.status !== 200) {
+          logger.error(
+            `${credentialName}: Authentication Failed. oidcToken field not detected in the response. Response = ${JSON.stringify(
+              result
+            )}`
+          );
+          throw new AuthenticationError(
+            response.status,
+            `${credentialName}: Authentication Failed. oidcToken field not detected in the response. Response = ${JSON.stringify(
+              result
+            )}`
+          );
+        } else {
+          logger.error(
+            `${credentialName}: Authentication Failed. oidcToken field not detected in the response but response status is 200.`
+          );
+          throw new AuthenticationError(
+            response.status,
+            `${credentialName}: Authentication Failed. oidcToken field not detected in the response but response status is 200.`
+          );
+        }
+      }
+    } catch (e) {
+      throw new AuthenticationError(
+        response.status,
+        `${credentialName}: Authentication Failed. oidcToken field not detected in the response. Response = ${text}`
```

Comment:
For this `catch` clause we should logger.error(e) so we do not lose the exception message